proposals/namelist_delimiters/namelist_proposal.rst

```diff
@@ -0,0 +1,107 @@
+:To: J3
+:From: Marshall Ward
+:Subject: Require delimiters for character arrays in namelist output
+:Date: 19 November 2019
+
+.. sectnum::
+
+Proposal for Fortran Standard: 202y
+
+
+Introduction
+============
+
+According to the current standard, a WRITE statement can write a namelist file
+that does not conform to the namelist specification.  This happens when the
+namelist group contains a character array and the DELIM specifier has a value
+of NONE.  In particular, this is the default behavior of a WRITE statement
+whose input is a namelist.
+
+Our proposal is to require delimiters when using WRITE to write a namelist to a
+file, by either requiring a value of DELIM which is namelist-compliant, or by
+overriding a value of NONE when the input is a namelist.
+
+
+Motivation
+==========
+
+The namelist format is described in section 13.11 of the standard, and
+13.11.3.3p7 requires that character arrays in namelist groups must be delimited
+with single or double quotes.
+
+   When the next effective item is of type character, the input form consists
+   of a sequence of zero or more rep-chars whose kind type parameter is implied
+   by the kind of the corresponding list item, delimited by apostrophes or
+   quotes.
+
+Any namelist whose character arrays are non-delimited is non-conformant.  Any
+parsing of this output is therefore considered to be unformatted, and the
+interpretation is at the discretion of the interpreter.
+
+Without delimiters, many character arrays become unparseable.  If a character
+array contains any lexical tokens, then it is likely that non-delimited values
+will be misinterpreted as part of the namelist object structure.  For example,
+any character array containing a backslash ``/`` may abruptly terminate the
```

Comment:
```suggestion
any character array containing a slash ``/`` may abruptly terminate the
```
Just to clarify, do you mean slash, or did you mean to put a backslash \ in here? Thanks!

Reply:
I did mean slash, thanks for the correction.